### ISSUE TITLE
Unexpected spacing when using `withStickyLogger`

### ISSUE THREAD

**Issue by joneshf** [commercialhaskell/rio#39]
I noticed a little weirdness when using `withStickyLogger`. The spacing before the `LogStr` behaves unexpectedly.

Here's some example code:

```Haskell
module Main where

import RIO

main :: IO ()
main = runRIO env $ do
  logInfo "info"
  logOther "other" "thing"

data Env
  = Env
    { envLogFunc :: !LogFunc
    }

instance HasLogFunc Env where
  logFuncL = to envLogFunc

env :: Env
env = Env { envLogFunc }
  where
  envLogFunc :: LogFunc
  envLogFunc cs source level str =
    withStickyLogger options $ \f -> f cs source level str
    where
    options :: LogOptions
    options = LogOptions
      { logMinLevel = LevelDebug
      , logTerminal = False
      , logUseColor = False
      , logUseTime = False
      , logUseUnicode = False
      , logVerboseFormat = True
      }
```

Will output:

```
[info] info
@(src/Main.hs:7:3)
[other]  thing
@(src/Main.hs:8:3)
```

The `LevelInfo` output seems correct–there is one space before the `LogStr`–but the `LevelOther` output seems incorrect–there are two spaces before the `LogStr`.

If you turn off verbose (`logVerboseFormat = False`), the output has one extra space at the beginning of the line:

```
 info
 thing
```

The extra space when `logVerboseFormat = False` seems to come from this line: https://github.com/commercialhaskell/rio/blob/0bbdf3b4d237e8bee3262c534b4716982dbd108d/src/RIO/Logger.hs#L208

The extra space for `LevelOther` when `logVerboseFormat = True` seems to come from this line: https://github.com/commercialhaskell/rio/blob/0bbdf3b4d237e8bee3262c534b4716982dbd108d/src/RIO/Logger.hs#L249

Seems like we could drop the space on line 208 and fix the problem if `getLevel` always appended a space for all the levels: https://github.com/commercialhaskell/rio/blob/0bbdf3b4d237e8bee3262c534b4716982dbd108d/src/RIO/Logger.hs#L241-L249

Seem fair/want a PR?

**Comment by joneshf** [comment]
Oh, I guess the intent could have been to have two spaces everywhere when `logVerboseFormat = True`. Either way, I think the solution is that same, move the spacing to `getLevel`. Just decide on how much spacing (I'm assuming one space).

**Comment by mgsloan** [comment]
:+1: for more consistent spacing.  PR would be appreciated!

**Comment by snoyberg** [comment]
Silly mistake on my part, should be fixed by bb88cc2. @joneshf can you confirm?

Thanks for the report.

**Comment by joneshf** [comment]
Looks great! Thanks!
